src/ansiblelint/transformer.py

```diff
@@ -0,0 +1,164 @@
+        return 0 if self.column < 2 else self._sequence_dash_offset
+
+    @sequence_dash_offset.setter
+    def sequence_dash_offset(self, value: int) -> None:
+        """Configure how many spaces to put before each sequence item's '-'."""
+        self._sequence_dash_offset = value
+
+
+# Transformer is for transforms like runner is for rules
+class Transformer:
+    """Transformer class performs the fmt transformations."""
+
+    def __init__(self, result: LintResult):
+        """Initialize a Transformer instance."""
+        # TODO: options for explict_start, indent_sequences
+        self.explicit_start = True
+        self.indent_sequences = True
+
+        self.matches: List[MatchError] = result.matches
+        self.files: Set[Lintable] = result.files
+
+        file: Lintable
+        lintables: Dict[str, Lintable] = {file.filename: file for file in result.files}
+        self.matches_per_file: Dict[Lintable, List[MatchError]] = {
+            file: [] for file in result.files
+        }
+
+        for match in self.matches:
+            try:
+                lintable = lintables[match.filename]
+            except KeyError:
+                # we shouldn't get here, but this is easy to recover from so do that.
+                lintable = Lintable(match.filename)
+                self.matches_per_file[lintable] = []
+            self.matches_per_file[lintable].append(match)
+
+    def run(self) -> None:
+        """Execute the fmt transforms."""
+        # ruamel.yaml rt=round trip (preserves comments while allowing for modification)
+        yaml = YAML(typ="rt")
+
+        # NB: ruamel.yaml does not have typehints, so mypy complains about everything here.
+
+        # configure yaml dump formatting
+        yaml.explicit_start = True  # type: ignore[assignment]
+        yaml.explicit_end = False  # type: ignore[assignment]
+
+        # TODO: make the width configurable
+        # yaml.width defaults to 80 which wraps longer lines in tests
+        yaml.width = 120  # type: ignore[assignment]
+
+        yaml.default_flow_style = False
+        yaml.compact_seq_seq = (  # dash after dash
+            True  # type: ignore[assignment]
+        )
+        yaml.compact_seq_map = (  # key after dash
+            True  # type: ignore[assignment]
+        )
+        # yaml.indent() obscures the purpose of these vars:
+        yaml.map_indent = 2  # type: ignore[assignment]
+        yaml.sequence_indent = 4 if self.indent_sequences else 2  # type: ignore[assignment]
+        yaml.sequence_dash_offset = yaml.sequence_indent - 2  # type: ignore[operator]
+
+        if self.indent_sequences:  # in the future: or other formatting options
+            # For root-level sequences, FormattedEmitter overrides sequence_indent
+            # and sequence_dash_offset to prevent root-level indents.
+            yaml.Emitter = FormattedEmitter
+
+        # explicit_start=True + map_indent=2 + sequence_indent=2 + sequence_dash_offset=0
```

Comment:
I think the information in this comment is valuable, but I don't think that an inline comment is a good place for it. I can't determine which line it "belongs" to. It feels like it's a [doctest](https://docs.python.org/3/library/doctest.html) for a method like `configure_yaml_formatter`

Reply:
I moved this into a docstring for a new `yaml` property. Does that help?

Reply:
I think that the comparison is a good way of explaining why all the fiddly knobs are the way they are. I saw you pushed it to the docstring, I think that's a good place for it. My personal style would be to extract a function and then have these as test cases. I find I usually don't regret breaking things up like that.

Reply:
So, test cases instead of or in addition to the docstring?

Reply:
I think if we intend for this configuration to be exposed publicly, we would probably need both (or doctests, which is already both). But if it's mostly internal, either is fine, I think it's a personal style thing. Even if we were planning on exposing this configuration publicly, I think we can safely push it off to another PR.

Reply:
I just added a parametrized test for this. :)